pkg/storegateway/chunkscache/memcached.go

```diff
@@ -0,0 +1,109 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+// Provenance-includes-location: https://github.com/thanos-io/thanos/blob/main/pkg/store/cache/memcached.go
+// Provenance-includes-license: Apache-2.0
+// Provenance-includes-copyright: The Thanos Authors.
```

Comment:
This is new code, no?

Reply:
meh, kind of. I copied it from pkg/storegateway/indexcache/memcached.go   which has this header 

https://github.com/grafana/mimir/blob/65bde07f57d1b3cdf7ffa54d5af9b244e2fca4dd/pkg/storegateway/indexcache/memcached.go#L1-L4

Reply:
I think it's different enough (chunks vs index, addition of pooling) that it counts as new code. @RichiH ?